tests/test_tempest_zigzag.py

```diff
@@ -17,21 +17,27 @@ def test_nothing_to_do(self, file_test_xml_all_pass, file_test_list):
         assert observed.attrib['errors'] == expected.attrib['errors']
         assert observed.attrib['tests'] == expected.attrib['tests']
         assert observed.attrib['name'] == 'xml suite created by tempest-zigzag'
-        assert len(observed) == len(expected)  # the actual number of child elements (testcase)
+        assert len(observed) == len(expected) + 1  # the actual number of child elements (testcase) + properties element
 
         for observed_case in observed:
             # test that the observed cases have the same names as the expected cases
-            assert len([case for case in expected if case.attrib['name'] == observed_case.attrib['name']]) is 1
+            try:
+                # there should be exactly one testcase with a matching name
+                assert len([case for case in expected if case.attrib['name'] == observed_case.attrib['name']]) == 1
+            except KeyError:  # <properties> do not have a name attribute
+                pass
 
-    def test_setupclass_failure(self, file_test_xml_setup_failure, file_test_list_with_whitespace):
+    def test_setupclass_failure(self, file_test_xml_setup_failure, file_test_list_with_whitespace, tempest_config_file):
         """Tests that the correct testcase elements will be created
         when a setUpClass failure is found"""
 
-        result = TempestZigZag.process_xml(file_test_xml_setup_failure, file_test_list_with_whitespace)
+        result = TempestZigZag.process_xml(file_test_xml_setup_failure,
+                                           file_test_list_with_whitespace,
+                                           tempest_config_file)
         observed = etree.XML(result)
         new_case_count = 0
 
-        assert len(observed) is 10
```

Comment:
Why did you switch from using 'is' to '==' for the assertions in these tests, just curious?

Reply:
pep8 now distinguishes what you are comparing, It now longer likes it when you use. `is` for comparison of integers.  🤷‍♂️

Reply:
:+1: